backend/internal/resource/handler.go

```diff
@@ -0,0 +1,734 @@
+/*
+ * Copyright (c) 2025, WSO2 LLC. (https://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package resource
+
+import (
+	"encoding/json"
+	"net/http"
+	"net/url"
+	"strconv"
+
+	serverconst "github.com/asgardeo/thunder/internal/system/constants"
+	"github.com/asgardeo/thunder/internal/system/error/apierror"
+	"github.com/asgardeo/thunder/internal/system/error/serviceerror"
+	"github.com/asgardeo/thunder/internal/system/log"
+	sysutils "github.com/asgardeo/thunder/internal/system/utils"
+)
+
+const handlerLoggerComponentName = "ResourceHandler"
+
+// resourceHandler handles HTTP requests for resource management.
+type resourceHandler struct {
+	resourceService ResourceServiceInterface
+}
+
+// newResourceHandler creates a new resource handler.
+func newResourceHandler(resourceService ResourceServiceInterface) *resourceHandler {
+	return &resourceHandler{
+		resourceService: resourceService,
+	}
+}
+
+// Resource Server Handlers
+
+// HandleResourceServerListRequest handles listing resource servers.
+func (h *resourceHandler) HandleResourceServerListRequest(w http.ResponseWriter, r *http.Request) {
+	logger := log.GetLogger().With(log.String(log.LoggerKeyComponentName, handlerLoggerComponentName))
+
+	limit, offset, svcErr := parsePaginationParams(r.URL.Query())
+	if svcErr != nil {
+		handleError(w, logger, svcErr)
+		return
+	}
+
+	result, svcErr := h.resourceService.GetResourceServerList(limit, offset)
+	if svcErr != nil {
+		handleError(w, logger, svcErr)
+		return
+	}
+
+	response := toResourceServerListResponse(result)
```

Comment:
WriteSuccessResponse/WriteErrorResponse helper functions were added with https://github.com/asgardeo/thunder/pull/845. Jihan is currently working on refactoring existing handlers. Shall we align this with that approach?